addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/WappalyzerPassiveScanner.java

```diff
@@ -162,6 +167,36 @@ private void checkMetaElementsMatches(Source source) {
         }
     }
 
+    private void checkDomElementMatches(Source source) {
+        Document doc = Jsoup.parse(String.valueOf(source));
+        for (Map<String, Map<String, Map<String, AppPattern>>> domSelectorMap :
+                currentApp.getDom()) {
+            for (Map.Entry<String, Map<String, Map<String, AppPattern>>> selectorMap :
+                    domSelectorMap.entrySet()) {
+                for (Map.Entry<String, Map<String, AppPattern>> nodeSelectorMap :
+                        selectorMap.getValue().entrySet()) {
+                    for (Map.Entry<String, AppPattern> value :
+                            nodeSelectorMap.getValue().entrySet()) {
+                        Elements selectedElements = doc.select(selectorMap.getKey());
+                        for (org.jsoup.nodes.Element selectedElement : selectedElements) {
+                            if (Objects.equals(value.getKey(), "text")) {
+                                AppPattern ap = value.getValue();
+                                addIfMatches(ap, selectedElement.text());
+                            }
+                            if (Objects.equals(nodeSelectorMap.getKey(), "attributes")
+                                    || Objects.equals(nodeSelectorMap.getKey(), "properties")) {
```

Comment:
There doesn't seem to be UnitTests that cover properties (with or without attributes).